plugins/lua/logistics.lua

```diff
@@ -6,94 +6,176 @@ local overlay = require('plugins.overlay')
 local utils = require('utils')
 local widgets = require('gui.widgets')
 
-local function make_stat(name, stockpile_number, stats, configs)
-    return {
-        enabled=configs[stockpile_number] and configs[stockpile_number][name] == 'true',
-        designated=stats[name..'_designated'][stockpile_number] or 0,
-        can_designate=stats[name..'_can_designate'][stockpile_number] or 0,
+-------------------------------------
+-- logistics status console output
+--
+
+local function make_stat(type, sp_number, stats, configs)
+    local stat = {
+        enabled=(configs[sp_number] ~= nil) and (configs[sp_number][type] == 'true'),
+        designated=stats[type..'_designated'][sp_number] or 0,
+        can_designate=stats[type..'_can_designate'][sp_number] or 0,
+        designatable=nil,
     }
+    stat.designatable = stat.designated + stat.can_designate
+    return stat
 end
 
 function getStockpileData()
     local stats, configs = logistics_getStockpileData()
-    local data = {}
-    for _,bld in ipairs(df.global.world.buildings.other.STOCKPILE) do
-        local stockpile_number, name = bld.stockpile_number, bld.name
-        local sort_key = tostring(name):lower()
-        if #name == 0 then
-            name = ('Stockpile #%d'):format(bld.stockpile_number)
-            sort_key = ('stockpile #%09d'):format(bld.stockpile_number)
+    local data = {sp_stats = {}, sp_totals = {}, sp_name_max_len = 0}
+    data.sp_totals = {
+        melt = {enabled_count = 0, designated = 0, designatable = 0},
+        trade = {enabled_count = 0, designated = 0, designatable = 0},
+        dump = {enabled_count = 0, designated = 0, designatable = 0},
+        train = {enabled_count = 0, designated = 0, designatable = 0},
+        forbid = {enabled_count = 0, designated = 0, designatable = 0},
+        claim = {enabled_count = 0}
```

Comment:
we generally do not add spaces around `=` inside of inline table declarations. that makes them visually distinct from regular assignment statements.